### ISSUE TITLE
Support more version types

### ISSUE THREAD

**Issue by hardikmodha** [bfred-it/tiny-version-compare#8]
Coming from https://github.com/sindresorhus/refined-github/pull/2042#issuecomment-492785932

1. Hotfixes: `1.0.0-hotfix` should be greater than `1.0.0` 
```
console.assert(
    tinyVersionCompare('1.0.0', '1.0.0-hotfix') === -1,
    '-1.0.0 should be lower'
);
```

2. GA (General Availability): `1.0.0-GA` should be greater than `1.0.0-M1`

```
console.assert(
    tinyVersionCompare('1.0.0-M1', '1.0.0-GA') === -1,
    '1.0.0-M1 should be lower'
);
```

3. RCs and M: `v1.0.0.RC1` should be greater than `v1.0.0.M2`.

```
console.assert(
    tinyVersionCompare('v1.0.0.RC1', 'v1.0.0.M1') === -1,
    'v1.0.0.M1 should be lower'
);
```

The above assertions fail. Can we add support for `hotfix`, `GA` and `M`?

**Comment by fregante** [comment]
1. is incorrect, at least as far as npm is concerned. 1.0.0-0 is considered a prerelease of 1.0.0, so I treat the dash as “it came before”

What’s M? Examples in the wild?

**Comment by hardikmodha** [comment]
I agree with you but can we treat numbers and string as different? e.g. `1.0.0-0` is a pre-release of `1.0.0` but `1.0.0-hotfix` is not and sorry for not being clear for `M`. It stands for milestone release. Kind of pre-alpha thing.

**Comment by notlmn** [comment]
https://github.com/github/version_sorter.

Just dropping this here in case anyone needs it, can't say if this is the one GitHub actually uses on their platform though.

**Comment by fregante** [comment]
Good one. I should probably find the most popular version sorting lib on npm and get its tests (but not just semver, because its specification is more restrictive than this module needs)
